### PR TITLE
Closes #45 Radio Buttons

### LINKED ISSUE #45: Radio buttons for filter types
We need radio buttons for filter types. LEEEGGGOOOOO

### PR DESCRIPTION
**Pull request by KyleNehman**
#45

Adds ability to setup radio buttons.

Known issue with the default selected value: @JoshuaStorm 
n Buttons have their values mapped from [0, n-1]. The call of value(0), which would be expected to highlight the button with value mapped to 0, does nothing. I'd assume this is an issue with the value 0 being equal to nothingness and the function interpreting it as null rather than the int value? 

My proposed solution would be to map the value of the buttons to something other than ints [0, n-1]

### CODE REVIEW

File: setupRadio.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
+function setupRadio(x, y, radioOptions, verticalFlag, defaultOption) {
+  var BUTTON_WIDTH = 30;
```
Comment (JoshuaStorm) [inline setupRadio.js]: Could you add a quick comment to just note this is in pixels (should be obvious, but y'know)

**Comment by JoshuaStorm** [comment]
Gimme a minute, I still don't understand this issue you're describing so I'll pull this branch and try it out myself.

File: setupRadio.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
+function setupRadio(x, y, radioOptions, verticalFlag, defaultOption) {
+  var BUTTON_WIDTH = 30;
+  var labelFontSize = floor(height / 75);
+  var thisRadio = createRadio();
+  var longestOptionLength = 0;
+  var radioWidth;
+
+  for (var i = 0; i < radioOptions.length; i++) {
+    var option = radioOptions[i];
```
Comment (JoshuaStorm) [inline setupRadio.js]: Wait so is `radioOptions` an array or?

File: setupRadio.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+      longestOptionLength = optionLength;
+    }
+  }
+  thisRadio.position(x, y);
+  if (verticalFlag) {
+    radioWidth = BUTTON_WIDTH + longestOptionLength;
+    thisRadio.style('width', radioWidth + 'px');
+  }
+
+  if(defaultOption) {
```
Comment (JoshuaStorm) [inline setupRadio.js]: You can actually do defaults in JavaScript with the fairly clever syntax of:

`var defaultVal = defaultOption || 1;`

File: setupRadio.js [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+  var labelFontSize = floor(height / 75);
+  var thisRadio = createRadio();
+  var longestOptionLength = 0;
+  var radioWidth;
+
+  for (var i = 0; i < radioOptions.length; i++) {
+    var option = radioOptions[i];
+    var optionLength = textWidth(option);
+
+    thisRadio.option(option, i);
```
Comment (JoshuaStorm) [inline setupRadio.js]: So, starting this at 0 does seem to create some issues. It seems to be a bug with `p5.js`, so if you could create a simplified example and report an issue on their GitHub I'm sure that'd get your name out there ;)

https://github.com/processing/p5.js/issues

For our use, we can just start at 1, most easily achieved by just making this line `thisRadio.option(option, i + 1)`

File: setupRadio.js [hunk 5]
```diff
@@ -0,0 +1,30 @@
+function setupRadio(x, y, radioOptions, verticalFlag, defaultOption) {
```
Comment (JoshuaStorm) [inline setupRadio.js]: The term `options` is generally reserved for associative objects in `js`. Maybe just `labelsArray` and `default`

**Comment by KyleNehman** [comment]
Counting starting at 1 feels so unnatural and weird, but I agree it's a simple fix until the folks over at P5 address the issue. All good now?

**Comment by JoshuaStorm** [comment]
Looks good!

**Pull request merged**